cmd/generator/generator.go

```diff
@@ -0,0 +1,339 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"os"
+	"path/filepath"
+	"slices"
+	"text/template"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+const (
+	defaultGoPermissions = 0644
+)
+
+var (
+	outputDirectory string
+	verbose         bool
+)
+
+type generatorDef struct {
+	Name        string
+	Type        string
+	DecodeFunc  string
+	EncoderFunc string
+	CheckerFunc string
+}
+
+func structToMap(def generatorDef) map[string]interface{} {
+	caser := cases.Title(language.English)
+
+	out := map[string]interface{}{
+		"Name":        caser.String(def.Name),
+		"Type":        def.Name,
+		"DecodeFunc":  def.DecodeFunc,
+		"EncoderFunc": def.EncoderFunc,
+		"CheckerFunc": def.CheckerFunc,
+	}
+
+	if def.Type != "" {
+		out["Type"] = def.Type
+	}
+
+	return out
+}
+
+var DefaultTypes = []generatorDef{
```

Comment:
Is it should be public? It confusing a bit.
```suggestion
var defaultTypes = []generatorDef{
```

Reply:
It's "main" module, so what is the problem?